stackit/internal/services/modelserving/token/resource.go

```diff
@@ -0,0 +1,665 @@
+				Description: "The time until the model serving auth token is valid.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *tokenResource) Create(
+	ctx context.Context,
+	req resource.CreateRequest, //nolint:gocritic // function signature required by Terraform
+	resp *resource.CreateResponse,
+) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from plan
+	var model Model
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectId := model.ProjectId.ValueString()
+	region := model.Region.ValueString()
+	if region == "" {
+		region = r.client.GetConfig().Region
+	}
+
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "region", region)
+
+	// Generate API request body from model
+	payload, err := toCreatePayload(&model)
+	if err != nil {
+		core.LogAndAddError(
+			ctx,
+			&resp.Diagnostics,
+			"Error creating model serving auth token",
+			fmt.Sprintf("Creating API payload: %v", err),
+		)
+		return
+	}
+
+	// Create new model serving auth token
+	createTokenResp, err := r.client.CreateToken(ctx, region, projectId).
+		CreateTokenPayload(*payload).
+		Execute()
+	if err != nil {
+		core.LogAndAddError(
+			ctx,
+			&resp.Diagnostics,
+			"Error creating model serving auth token",
+			fmt.Sprintf("Calling API: %v", err),
+		)
+		return
+	}
+
+	waitResp, err := wait.CreateModelServingWaitHandler(ctx, r.client, region, projectId, *createTokenResp.Token.Id).
+		WaitWithContext(ctx)
+	if err != nil {
+		core.LogAndAddError(
+			ctx,
+			&resp.Diagnostics,
+			"Error creating model serving auth token",
+			fmt.Sprintf("Waiting for token to be active: %v", err),
+		)
+		return
+	}
+
+	// Map response body to schema
+	err = mapCreateResponse(createTokenResp, waitResp, &model)
+	if err != nil {
+		core.LogAndAddError(
+			ctx,
+			&resp.Diagnostics,
+			"Error creating model serving auth token",
+			fmt.Sprintf("Processing API payload: %v", err),
+		)
+		return
+	}
+
+	// Set state to fully populated data
+	diags = resp.State.Set(ctx, model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Info(ctx, "Model-Serving auth token created")
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *tokenResource) Read(
+	ctx context.Context,
+	req resource.ReadRequest, //nolint:gocritic // function signature required by Terraform
+	resp *resource.ReadResponse,
+) { // nolint:gocritic // function signature required by Terraform
+	var model Model
+	diags := req.State.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectId := model.ProjectId.ValueString()
+	tokenId := model.TokenId.ValueString()
+	region := model.Region.ValueString()
+	if region == "" {
+		region = r.client.GetConfig().Region
+	}
```

Comment:
```suggestion
	var region string
	if utils.IsUndefined(model.Region) {
		region = r.providerData.GetRegion()
	} else {
		region = model.Region.ValueString()
	}
```

The region of the API client (from the SDK) will be removed long-term due to the implementation of the multi-region concept. So please get the default value from the provider config instead.

Reply:
Referenced utils pkg: `github.com/stackitcloud/terraform-provider-stackit/stackit/internal/utils`